### PR TITLE
Chromium: Implement Launcher.userDataDirTests

### PR DESCRIPTION
**Pull request by kblok**
related to #68

### CODE REVIEW

File: src/PlaywightSharp.ProtocolTypesGenerator/Chromium/ChromiumProtocolTypesGenerator.cs [hunk 1]
```diff
@@ -99,13 +99,13 @@ private void PrepareArrayTypes(ChromiumProtocolDomainsContainer response)
                     }
                     else if (type.Type == "integer")
                     {
-                        _knownTypes[type.Id] = "int";
-                        _knownTypes[$"{domain.Domain}.{type.Id}"] = "int";
+                        _knownTypes[type.Id] = "int?";
+                        _knownTypes[$"{domain.Domain}.{type.Id}"] = "int?";
```
Comment (kblok) [inline src/PlaywightSharp.ProtocolTypesGenerator/Chromium/ChromiumProtocolTypesGenerator.cs]: Properties need to be nullable, if we don't do this we would send properties with primitives default values, such as 0 for timeout

File: src/PlaywrightSharp.Chromium/ChromiumExecutionContext.cs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        public ChromiumExecutionContext(ChromiumSession client, ExecutionContextDescription contextPayload)
+        {
+            _client = client;
+            _contextId = contextPayload.Id.Value;
+        }
+
+        public async Task EvaluateAsync(FrameExecutionContext context, bool returnByValue, string script, object[] args)
+            => await EvaluateAsync<object>(context, returnByValue, script, args).ConfigureAwait(false);
+
+        public async Task<T> EvaluateAsync<T>(FrameExecutionContext context, bool returnByValue, string script, object[] args)
```
Comment (kblok) [inline src/PlaywrightSharp.Chromium/ChromiumExecutionContext.cs]: This code is an MVP for what we need to do in the PR.

File: src/PlaywrightSharp.Chromium/Protocol/RemoteObjectType.cs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+namespace PlaywrightSharp.Chromium.Protocol
+{
+    internal static class RemoteObjectType
+    {
+        public static string Bigint => "bigint";
```
Comment (kblok) [inline src/PlaywrightSharp.Chromium/Protocol/RemoteObjectType.cs]: We might get more here. if we don't we can move it to a const.

File: src/PlaywrightSharp.Tests/Launcher/UserDataDirTests.cs [hunk 4]
```diff
@@ -70,21 +72,25 @@ public async Task UserDataDirOptionShouldRestoreState()
...
-            using (var browser2 = await Playwright.LaunchAsync(options))
-            {
-                var page2 = await browser2.DefaultContext.NewPageAsync();
-                await page2.GoToAsync(TestConstants.EmptyPage);
-                Assert.Equal("hello", await page2.EvaluateAsync<string>("localStorage.hey"));
-            }
+            var page = await browser.DefaultContext.NewPageAsync();
+            await page.GoToAsync(TestConstants.EmptyPage);
+            await page.EvaluateAsync("localStorage.hey = 'hello'");
+            await browser.CloseAsync();
```
Comment (kblok) [inline src/PlaywrightSharp.Tests/Launcher/UserDataDirTests.cs]: Dispose calls are fire and forget, we need to await for the close here.

Comment (Meir017) [reply src/PlaywrightSharp.Tests/Launcher/UserDataDirTests.cs]: maybe we should implement `IAsyncDisposable`?

Comment (kblok) [reply src/PlaywrightSharp.Tests/Launcher/UserDataDirTests.cs]: The issue with IAsyncDisposable is that it's .NET standard 2.1
That would leave .NET Framework out. I will try to see if I can make it multi-target in another PR.

**Pull request merged**